charts/opentelemetry-collector/templates/_helpers.tpl

```diff
@@ -68,8 +68,10 @@ app.kubernetes.io/component: agent-collector
 {{- if eq .Values.mode "statefulset" }}
 app.kubernetes.io/component: statefulset-collector
 {{- end -}}
+{{- if .Values.additionalLabels }}
```

Comment:
opentelemetry-collector.additionalLabels already includes an if statement so I suspect the actual issue is that this `include` doesn't have a leading `-`.

Can you test with `{{- include "opentelemetry-collector.additionalLabels" . }}`

Reply:
ups, didn't see the other `if` statement 🤦🏾
 
But a leading `-` doesn't fix it. From what i understand, the ´{{- ´ is supposed to remove any whitespace from the left, so it would do the opposite.

To avoid the duplicate the duplicate `if`, I can add  `| nindent 0` to either `{{- tpl (.Values.additionalLabels | toYaml) . }}` or  `{{ include "opentelemetry-collector.additionalLabels" . }}` which works just fine, but looks more like a "hack"

I can also remove the leading ´-´ from `{{ tpl (.Values.podLabels | toYaml) . }}`, which also works, but is not consistent with the other usages of `{{- tpl() }}` in the file. 

I'm not a Go/Helm template expert, so this is what i got :)

Thanks for review btw! 🙏🏾

Reply:
Looking through the chart, we only use `opentelemetry-collector.additionalLabels` inside `opentelemetry-collector.labels`.  Lets remove `opentelemetry-collector.additionalLabels` and move its logic directly into 
opentelemetry-collector.labels`. It should look like

```
{{/*
Common labels
*/}}
{{- define "opentelemetry-collector.labels" -}}
helm.sh/chart: {{ include "opentelemetry-collector.chart" . }}
{{ include "opentelemetry-collector.selectorLabels" . }}
{{- if .Chart.AppVersion }}
app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
{{- end }}
app.kubernetes.io/managed-by: {{ .Release.Service }}
{{- if eq .Values.mode "deployment" }}
app.kubernetes.io/component: standalone-collector
{{- end -}}
{{- if eq .Values.mode "daemonset" }}
app.kubernetes.io/component: agent-collector
{{- end -}}
{{- if eq .Values.mode "statefulset" }}
app.kubernetes.io/component: statefulset-collector
{{- end -}}
{{- if .Values.additionalLabels }}
{{ tpl (.Values.additionalLabels | toYaml) . }}
{{- end }}
{{- end }}
```